tools/SourceKit/tools/sourcekitd/lib/API/VariableTypeArray.cpp

```diff
@@ -0,0 +1,191 @@
+//===---------- VariableTypeArray.cpp -------------------------------------===//
```

Comment:
I think it’s kind of unfortunate that we need to duplicate so much implementation between `ExpressionTypeArray` and `VariableTypeArray`. Do you think it would be feasible to share some implementation between the two? It might not be possible, I didn’t check…

Reply:
I agree, that's a bit unfortunate. Sharing the implementation might be a bit tricky though, as many parts of this implementation make close assumptions about how `VariableType` and `ExpressionType` are structured and would require a fair bit of templating to abstract over these differences, from what I can tell.

Reply:
If that’s the case, I think that duplicating is indeed the best option…